Gitbucket 4.35.0 release announcement (#53)
gitbucket/gitbucket-news: _posts/2020-12-25-gitbucket-4.35.0.markdown

image: /images/gitbucket-4.35.0/color_theme_editor.png
categories: gitbucket
---

Today, we are pleased to announce the availability of new version of Scala based Git centric software development platform [GitBucket 4.35.0](https://github.com/gitbucket/gitbucket/releases/tag/4.35.0)!

Since this release includes various new features, improvements and bug fixes, it's really challenging to cover all changes in this document. For now, let's see the major stuff in this release.

## Editor and source viewer color theme

The color theme of the online editor and the source code viewer became selectable.

![Color theme of the editor]({{site.baseurl}}/images/gitbucket-4.35.0/color_theme_editor.png)

The source code viewer can be confiregured in your account settings.

![Color theme of the source viewer]({{site.baseurl}}/images/gitbucket-4.35.0/color_theme_viewer.png)

## Auto completion for issues and pull requests

Auto completion now supports issues and pull requests. You can trigger the auto completion by typing `#`.

![Auto completion for issues and pull requests]({{site.baseurl}}/images/gitbucket-4.35.0/auto_completion_issue.png)

## Upload image from clipboard

> GitBucket now supports uplaading images from clipboard. You can paste (`CTRL + V`) an image from clipboard whereever you can upload files on GitBucket (e.g. issues, pull requests, comments, etc).

Reviewer: ```suggestion
GitBucket now supports uploading images from clipboard. You can paste (`CTRL + V`) an image from clipboard wherever you can upload files on GitBucket (e.g. issues, pull requests, comments, etc).
```

---
layout: post
title: "GitBucket 4.35.0 released!"
date: 2020-12-25 00:00:00
image: /images/gitbucket-4.35.0/color_theme_editor.png
categories: gitbucket
---

Today, we are pleased to announce the availability of new version of Scala based Git centric software development platform [GitBucket 4.35.0](https://github.com/gitbucket/gitbucket/releases/tag/4.35.0)!

> Since this release includes various new features, improvements and bug fixes, it's really challenging to cover all changes in this document. For now, let's see the major stuff in this release.

Reviewer: ```suggestion
Since this release includes various new features, improvements, and bug fixes, it's really challenging to cover all changes in this document. For now, let's see the major stuff in this release.
```

## Editor and source viewer color theme

The color theme of the online editor and the source code viewer became selectable.

![Color theme of the editor]({{site.baseurl}}/images/gitbucket-4.35.0/color_theme_editor.png)

> The source code viewer can be confiregured in your account settings.

Reviewer: ```suggestion
The source code viewer can be configured in your account settings.
```

![Color theme of the source viewer]({{site.baseurl}}/images/gitbucket-4.35.0/color_theme_viewer.png)

## Auto completion for issue number

Auto completion now supports issue number.

![Auto completion for issue number]({{site.baseurl}}/images/gitbucket-4.35.0/auto_completion_issue.png)

## Upload image from clipboard

> GitBucket now supports uplaading images from clipboard. You can paste (CTRL + V) an image from clipboard whereever you can upload files on GitBucket (e.g. issues, pull requests, comments, etc). We tested this features on the following environments:

Reviewer: ```suggestion
GitBucket now supports uploading images from clipboard. You can paste (CTRL + V) an image from clipboard wherever you can upload files on GitBucket (e.g. issues, pull requests, comments, etc.). We tested this features on the following environments:
```

- Chrome and Firefox on Windows 10
- Chrome and Firefox on MacOS Catalina

This feature requires Clipboard API, so it doesn't work with legacy browsers like Internet Explorer that don't implement Clipboard API.

## Close multiple issues by commit comment

To date, we needed to write a commit comment to close multiple issues at once.

```
fix #1, fix #2, fix #3
```

In GitBucket 4.35.0, we now can write this as follows:

```
fix #1, #2, #3
```

## Create pull request from online editor

When we edit a file in Git repository using the online editor, we can create a pull request instead of commit to the branch. Even users who don't have a permission to commit to the repository can create a pull request using the online editor.

![Create pull request from the editor]({{site.baseurl}}/images/gitbucket-4.35.0/create_pull_request.png)

> ## Milstone overview

Reviewer: ```suggestion
## Milestone overview
```

> GitBucket now provides a desicated page for the milestone overview. This page shows all issues and pull requests that are associated with the milestone at once.

Reviewer: ```suggestion
GitBucket now provides a dedicated page for the milestone overview. This page shows all issues and pull requests that are associated with the milestone at once.
```

![Milestone overview]({{site.baseurl}}/images/gitbucket-4.35.0/milestone.png)

## Commit status at various places

> If you use CI integartion with GitBucket, the build status is shown but available places were very limited. In this release, the build status is shown at various places like the branch list and the commit detailed page. Also, the status of each context is now shown in the tooltip dialog.

Reviewer: ```suggestion
If you use CI integration with GitBucket, the build status is shown but available places were very limited. In this release, the build status is shown at various places like the branch list and the commit detailed page. Also, the status of each context is now shown in the tooltip dialog.
```
